### DISCUSSION TITLE
Have 64-bit operating system but PowerShell using 32-bit system

### DISCUSSION THREAD

**Discussion by kalirs256** [PowerShell/PowerShell#16769] [Q&A]
Hi y'all,

Forgive me if the answer is really obvious since I have pretty much zero experience with programming. I'm trying to work with a program through PowerShell but have been running into a bunch of issues. I'm wondering if it's because the program only works with 64-bit operating systems. When I go to my computer settings, it says that I'm running a 64-bit operating system, x64-based processor. But when I use PowerShell it says `PS C:\WINDOWS\system32>`. I'm wondering if this is what's causing the issue. Is there a way to update PowerShell to a 64-bit version?

Thanks!

**Comment by vexx32** [comment]
Windows retained `system32` as the default system directory, despite running on `x64` chips these days -- things in there are typically 64-bit on a 64-bit Windows installation. Unsure of the reason, there's probably some historical reason for it.

On 64-bit windows there's a `SysWOW64` directory that contains all the real 32-bit assemblies, and Windows does... something weird (unsure of details)... to allow 32-bit applications to "see" that folder as `C:\Windows\system32` even though it's at a different location. So... yeah, no, you'll see system32 on every Windows system these days, even if it's a 64-bit system.

Not sure if we're still shipping 32-bit versions of pwsh to be honest, but if we are you can check the installer you grabbed to see what it's installing -- the x64 installers are labelled as such, just make sure you don't try to use the `arm64` ones, those are for (mainly) mobile devices and a few others that ship with ARM processors.
